uiautomationutilities/perf_tool/PerfData.cs

```diff
@@ -0,0 +1,202 @@
+                throw new Exception($"File not found {filePath}");
+            }
+            PerfDataRecordsList = File.ReadLines(filePath)
+                .Select(line => line.Split(','))
+                .Select(tokens => new PerfDataRecord(
+                    tokens[MapHeaderToIndex("TimeStamp")],
+                    tokens[MapHeaderToIndex("Marker")],
+                    tokens[MapHeaderToIndex("Time")],
+                    tokens[MapHeaderToIndex("Thread")],
+                    tokens[MapHeaderToIndex("CpuUsed")],
+                    tokens[MapHeaderToIndex("CpuTotal")],
+                    tokens[MapHeaderToIndex("ResidentSize")],
+                    tokens[MapHeaderToIndex("VirtualSize")],
+                    tokens[MapHeaderToIndex("WifiSent")],
+                    tokens[MapHeaderToIndex("WifiRecv")],
+                    tokens[MapHeaderToIndex("WwanSent")],
+                    tokens[MapHeaderToIndex("WwanRecv")],
+                    tokens[MapHeaderToIndex("AppSent")],
+                    tokens[MapHeaderToIndex("AppRecv")],
+                    tokens[MapHeaderToIndex("Battery")],
+                    tokens[MapHeaderToIndex("SystemDiskRead")],
+                    tokens[MapHeaderToIndex("SystemDiskWrite")]
+                    )
+                ).ToList();
+            //Check if header is present
+            if (PerfDataRecordsList.ElementAt(0).TimeStamp.Equals("TimeStamp"))
+            {
+                _headers = PerfDataRecordsList.ElementAt(0);
+                PerfDataRecordsList.RemoveAt(0);
+            }
+        }
+
+        private int MapHeaderToIndex(string token)
+        {
+            switch(token)
+            {
+                case "TimeStamp": return 0;
+                case "Marker": return 1;
+                case "Time": return 2;
+                case "Thread": return 3;
+                case "CpuUsed": return 4;
+                case "CpuTotal": return 5;
+                case "ResidentSize": return 6;
+                case "VirtualSize": return 7;
+                case "WifiSent": return 8;
+                case "WifiRecv": return 9;
+                case "WwanSent": return 10;
+                case "WwanRecv": return 11;
+                case "AppSent": return 12;
+                case "AppRecv": return 13;
+                case "Battery": return 14;
+                case "SystemDiskRead": return 15;
+                case "SystemDiskWrite": return 16;
+                default: return -1;
+            }
+        }
+
+        public PerfDataRecord FindMarker(String marker, int skipCount = 0)
+        {
+            var records = PerfDataRecordsList.Where(t => t.Marker.Equals(marker, StringComparison.InvariantCultureIgnoreCase));
+            if (records.Count() > skipCount)
+            {
+                return records.ElementAt(skipCount);
+            }
+            else
+            {
+                return null;
+            }
+        }
+        public void AddMarkerNames()
```

Comment:
```suggestion

        public void AddMarkerNames()
```

General formatting preference: let's add some line breaks between functions to keep the source easily visually scannable. Public methods to get documentation or have their visibility reduced (if they shouldn't be public)